ibm/resource_ibm_is_flow_log.go

```diff
@@ -0,0 +1,407 @@
+			},
+
+			ResourceControllerURL: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The URL of the IBM Cloud dashboard that can be used to explore and view details about this instance",
+			},
+
+			ResourceName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The name of the resource",
+			},
+
+			ResourceCRN: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The crn of the resource",
+			},
+
+			ResourceStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the resource",
+			},
+
+			ResourceGroupName: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The resource group name in which resource is provisioned",
+			},
+		},
+	}
+}
+
+func resourceIBMISFlowLogValidator() *ResourceValidator {
+
+	validateSchema := make([]ValidateSchema, 1)
+
+	validateSchema = append(validateSchema,
+		ValidateSchema{
+			Identifier:                 isFlowLogName,
+			ValidateFunctionIdentifier: ValidateRegexpLen,
+			Type:                       TypeString,
+			Required:                   true,
+			Regexp:                     `^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$`,
+			MinValueLength:             1,
+			MaxValueLength:             63})
+
+	ibmISFlowLogValidator := ResourceValidator{ResourceName: "ibm_is_flow_log", Schema: validateSchema}
+	return &ibmISFlowLogValidator
+}
+
+func resourceIBMisFlowLogCreate(d *schema.ResourceData, meta interface{}) error {
+	sess, err := vpcClient(meta)
+	if err != nil {
+		return err
+	}
+
+	createFlowLogCollectorOptionsModel := &vpcv1.CreateFlowLogCollectorOptions{}
+	name := d.Get(isFlowLogName).(string)
+	createFlowLogCollectorOptionsModel.Name = &name
+	if _, ok := d.GetOk(isFlowLogResourceGroup); ok {
+		group := d.Get(isFlowLogResourceGroup).(string)
+		resourceGroupIdentityModel := new(vpcv1.ResourceGroupIdentityByID)
+		resourceGroupIdentityModel.ID = &group
+		createFlowLogCollectorOptionsModel.ResourceGroup = resourceGroupIdentityModel
+	}
+	if _, ok := d.GetOk(isFlowLogActive); ok {
+		active := d.Get(isFlowLogActive).(bool)
+		createFlowLogCollectorOptionsModel.Active = &active
+	}
+
+	target := d.Get(isFlowLogTarget).(string)
+	FlowLogCollectorTargetModel := &vpcv1.FlowLogCollectorPrototypeTarget{}
+	FlowLogCollectorTargetModel.ID = &target
+	createFlowLogCollectorOptionsModel.Target = FlowLogCollectorTargetModel
+
+	bucketname := d.Get(isFlowLogStorageBucket).(string)
+	cloudObjectStorageBucketIdentityModel := new(vpcv1.CloudObjectStorageBucketIdentityByName)
+	cloudObjectStorageBucketIdentityModel.Name = &bucketname
+	createFlowLogCollectorOptionsModel.StorageBucket = cloudObjectStorageBucketIdentityModel
+
+	flowlogCollector, response, err := sess.CreateFlowLogCollector(createFlowLogCollectorOptionsModel)
+	if err != nil {
+		log.Printf("[DEBUG] Create Flow Log Collector err %s\n%s", err, response)
```

Comment:
Make all return stmt to include both response and error...which we did for pdns resources